biolearn/data/library.yaml

```diff
@@ -1,5 +1,190 @@
 ---
 items:
+  - id: GSE112618
+    title: 'FACS validation dataset: An optimized library for reference-based deconvolution
+      of whole-blood biospecimens assayed using the Illumina HumanMethylationEPIC BeadArray
+      (III)'
+    summary: DNA methylation assessments of peripheral blood DNA can be used to accurately
+      estimate the relative proportions of underlying leukocyte subtypes. Such cell
+      deconvolution analysis relies on libraries of discriminating differentially methylated
+      regions that are developed for each specific cell type measured. The relationship
+      between estimated cell type proportions can then be tested for their association
+      with phenotypes, disease states, and subject outcomes, or used in multivariable
+      models as terms for adjustment in epigenome-wide association studies (EWAS). We
+      obtained purified neutrophils, monocytes, B-lymphocytes, natural killer (NK) cells,
+      CD4+ T-cells, and CD8+ T-cells from healthy subjects and measured DNA methylation
+      with the Illumina HumanMethylationEPIC array platform. In addition, we measured
+      DNA methylation with the EPIC array in two sets of artificial DNA mixtures comprising
+      the above cell types. We compared three separate approaches to select reference
+      differentially methylated region libraries (DMR library), for cell type proportion
+      inference. The IDOL algorithm identified an optimal DMR library consisting of
+      450 CpG sites for inferring leukocyte subtype proportions (average R2=99.2). Importantly,
+      the majority of CpG sites (69%) in the IDOL DMR library were unique to the new
+      EPIC methylation array, in that they were not present on the 450K array. Our new
+      reference DMR library is available as a Bioconductor package, has the potential
+      to reduce any unintended technical differences arising from the combination of
+      different generations of array platforms, and may be helpful in generating larger
+      DMR libraries that include novel cell subtypes. A  dataset of six whole blood
+      samples were FACS sorted and the DNA was processed as a validation dataset.
+    format: Unknown
+    organism: human
+    samples: 6
+    path: https://ftp.ncbi.nlm.nih.gov/geo/series/GSE112nnn/GSE112618/matrix/GSE112618_series_matrix.txt.gz
+    parser:
+      type: geo-matrix
+      id_row: null
+      matrix-start: 82
+      metadata:
+        sample_well:
+          row: 45
+          parse: string
+        chip:
+          row: 46
+          parse: string
+        sex:
+          row: 47
+          parse: string
```

Comment:
Use "sex" as the parse function. Assuming this parse function will return a good result: https://github.com/bio-learn/biolearn/blob/2e975f9042050dd65907682043aa7c1558d16190/biolearn/data_library.py#L18